py/persistentcode.c

```diff
@@ -221,10 +253,19 @@ STATIC mp_obj_t load_obj(mp_reader_t *reader) {
         return MP_OBJ_FROM_PTR(&mp_const_ellipsis_obj);
     } else {
         size_t len = read_uint(reader);
+
         if (len == 0 && obj_type == 'b') {
             read_byte(reader); // skip null terminator
             return mp_const_empty_bytes;
         }
+
+        #if MICROPY_VFS_MAP
+        if (reader_is_map(reader)) {
+            const uint8_t *data = map_read_bytes(reader, len + 1);
+            return mp_obj_new_str_static(obj_type == 's' ? &mp_type_str : &mp_type_bytes, data, len);
```

Comment:
If it gets here, it doesn't handle integers, floats, and complex types properly because it returns right away.

To reproduce, run a script containing:

```python
-1.0
```

This gives: `TypeError: unsupported type for __neg__: 'bytes'`

When rebasing this, it could be fixed by combining it with the parts that handle int/float/complex below. Here is a [non-optimized fix](https://github.com/pybricks/micropython/commit/5f416948325a616bb937331997177f33543953fe#diff-3ddb4f24934bdf20e222654d987dec0db7e6d67da578d76895cd71d1a3f82a0eR232-R245).

Reply:
Thanks, now rebased on latest master and this issue fixed.